File: android/app/src/main/java/jagtheora/vorbis/DSPState.java
package jagtheora.vorbis;

import jagtheora.misc.SimplePeer;

public final class DSPState extends SimplePeer {

    public DSPState(VorbisInfo arg0) {
        this.init(arg0);
        if (this.b()) {
            throw new IllegalStateException();
        }
    }

    public float[][] pcmOut(int arg0) { return null; }

    public void init(VorbisInfo arg0) {}

    public double granuleTime() { return 0d; }

    protected void clear() {}

    public void blockIn(VorbisBlock arg0) {}

    public void read(int arg0) {}
}

File: src/jvmMain/kotlin/jagtheora/vorbis/DSPState.kt
package jagtheora.vorbis

import jagtheora.misc.SimplePeer

actual class DSPState actual constructor(arg0: VorbisInfo?) : SimplePeer() {
    init {
        this.init(arg0)
        check(!this.b())
    }

    actual external fun pcmOut(arg0: Int): Array<FloatArray?>?

    /*private*/ actual external fun init(arg0: VorbisInfo?)

    actual external fun read(arg0: Int)

    actual external fun blockIn(arg0: VorbisBlock?)

    actual external override fun clear()

    actual external fun granuleTime(): Double
}
